### DISCUSSION TITLE
Pass query parameters to runOnDemand commands

### DISCUSSION THREAD

**Discussion by mercora** [bluenviron/mediamtx#329] [General]
## Which version are you using?

r557.93ba061-1

## Enhancement request
It would be nice if it would be possible to pass the full URL given to the server to runOnDemand commands as i would like to parse the query parameters of a given URL in order to react on them. Ideally it would be possible to get the full URL given to the server but any method to pass on query parameters would be fine. With sufficiently helping hints i might be able to implement this by myself but currently i feel a little lost. it looks like the environment gets initialized [here](https://github.com/aler9/rtsp-simple-server/blob/93ba0612777c2daf0c494e9c1ca1a33ebcbc1a5e/internal/path/path.go#L555) but i am not able to determine how i would be able to add the query parameters there. How is this initialized when using regexp matches?

**Comment by aler9** [comment]
Hello, this could be implemented but it's difficult - paths are initialized and stored exclusively by name, the query part of the request is discarded.

This is implemented in this way to allow different clients to use the same path, regardless of the query parameters. Keep in mind that some clients automatically add query parameters.

**Comment by EmadHelmi** [comment]
@mercora Did you find any solutions for this? I, also, want to have the query parameters in the path, because my ffmpeg command in the RunOnReady should be able to pass them to the externalAuth service. Otherwise, the RunOnReady command fails.

**Reply by aler9** [reply]
This discussion is really old, there's a feature request to add additional environment variables to runOn* command here: #1414

[a comment by an automated account is not shown]
